src/Rest/GreenhouseRoute.php

```diff
@@ -0,0 +1,668 @@
+	 *
+	 * @throws MissingFilterInfoException When not all required keys are set.
+	 *
+	 * @return void
+	 */
+	private function verifyGreenhouseConfirmationInfoExists(): void
+	{
+		if (empty(\apply_filters(Filters::GREENHOUSE_CONFIRMATION, 'name'))) {
+			throw MissingFilterInfoException::viewException(Filters::GREENHOUSE_CONFIRMATION, 'name');
+		}
+
+		if (empty(\apply_filters(Filters::GREENHOUSE_CONFIRMATION, 'email'))) {
+			throw MissingFilterInfoException::viewException(Filters::GREENHOUSE_CONFIRMATION, 'email');
+		}
+
+		if (empty(\apply_filters(Filters::GREENHOUSE_CONFIRMATION, 'subject'))) {
+			throw MissingFilterInfoException::viewException(Filters::GREENHOUSE_CONFIRMATION, 'subject');
+		}
+
+		// Returning value can't be mocked via tests.
+		if (defined('IS_TEST') && ! IS_TEST) {
+			if (apply_filters(Filters::GREENHOUSE_CONFIRMATION, 'name') === 'name') {
+				throw MissingFilterInfoException::viewException(Filters::GREENHOUSE_CONFIRMATION, 'name');
+			}
+
+			if (apply_filters(Filters::GREENHOUSE_CONFIRMATION, 'email') === 'email') {
+				throw MissingFilterInfoException::viewException(Filters::GREENHOUSE_CONFIRMATION, 'email');
+			}
+
+			if (apply_filters(Filters::GREENHOUSE_CONFIRMATION, 'subject') === 'subject') {
+				throw MissingFilterInfoException::viewException(Filters::GREENHOUSE_CONFIRMATION, 'subject');
+			}
+		}
+	}
+
+	/**
+	 * Make sure we have the data we need defined as filters for the fallback email.
+	 *
+	 * @throws MissingFilterInfoException When not all required keys are set.
+	 *
+	 * @return void
+	 */
+	private function verifyGreenhouseFallbackInfoExists(): void
+	{
+		if (empty(\apply_filters(Filters::GREENHOUSE_FALLBACK, 'email'))) {
+			throw MissingFilterInfoException::viewException(Filters::GREENHOUSE_FALLBACK, 'email');
+		}
+
+		if (empty(\apply_filters(Filters::GREENHOUSE_FALLBACK, 'subject'))) {
+			throw MissingFilterInfoException::viewException(Filters::GREENHOUSE_FALLBACK, 'subject');
+		}
+
+		// Returning value can't be mocked via tests.
+		if (defined('IS_TEST') && ! IS_TEST) {
+			if (apply_filters(Filters::GREENHOUSE_CONFIRMATION, 'email') === 'email') {
+				throw MissingFilterInfoException::viewException(Filters::GREENHOUSE_FALLBACK, 'email');
+			}
+
+			if (apply_filters(Filters::GREENHOUSE_CONFIRMATION, 'subject') === 'subject') {
+				throw MissingFilterInfoException::viewException(Filters::GREENHOUSE_FALLBACK, 'subject');
+			}
+		}
+	}
+
+	/**
+	 * Defines a list of required parameters which must be present in the request as GET parameters or it will error out.
+	 *
+	 * @return array
+	 */
+	protected function getRequiredPostParams(): array
+	{
+		return [
+			self::EMAIL_PARAM,
+			self::FIRST_NAME_PARAM,
+			self::LAST_NAME_PARAM,
+			self::JOB_ID_PARAM,
+		];
+	}
+
+	/**
+	 * Returns allowed methods for this route.
+	 *
+	 * @return string|array
+	 */
+	protected function getMethods()
+	{
+		return static::CREATABLE;
+	}
+}
```

Comment:
Couldn't find anything wrong with this other than it's breaking the single responsibility principle big time. This class should handle just the REST route, everything else (sending emails, validation, etc.) should have been a different class that does just that and injected here.

I won't require any rewrite because that would take too much time, but just something to be aware of.